### PR TITLE
Break on TypeScript parameter properties

### LINKED ISSUE #5666: TypeScript constructor assignment improvement.
**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACMAGwEMBnUvAFTlJgGETy9gAdKPA6W7AVzBhwAKNhw4ZsASwBuxeHmIYJSPAEFFAZTjYpEsHAA0I0eOmy4nKADMJAc2V1o1m5u264RgJTMAvm28h9EAgMGAkuZFBibGwIAHcABSiEUmQQYikICQATAJAAI2xiMABrOBh1DCKJKBtkGF4DEAALGABbQgB1Jol4Ukq9TShSHukegE9UsHJc6tItGHjCm1biZEtiQjnAgCtSNAAhQpKy9WJWuAAZarg1ja2QXbR1aptCOABFHgh4W83Gyuwc2wqRgYwwNDAkhCuRMsA62RgTWQAA4AAyBcQQOYdQoYVLiGhaKQ3QLYOAARx4EjJi2Iy1WSHWf0Cc1aEjqDRZLzen2+N0Zd0aMGIeXhWURyAATIF6sQJIQXg5WitUlBoCSQDw5pQRSkBX9vN4gA)
```sh
--parser typescript
```

**Input:**
```tsx
export class TestClass {
  constructor(
    private api: ApiService,
    private config: ConfigService
  ) {}
}
```

**Output:**
```tsx
export class TestClass {
  constructor(private api: ApiService, private config: ConfigService) {}
}

```

**Expected behavior:**

Consider breaking line despite `printWidth` value (as input).
Since it visually should be close to:

```tsx
export class TestClass {
  private api: ApiService;
  private config: ConfigService;
}
```

and not:

```tsx
export class TestClass {
  private api: ApiService; private config: ConfigService;
}
```

### PR DESCRIPTION
**Pull request by seiyab**
## Description
This addresses https://github.com/prettier/prettier/issues/5666 .

Fixes #5666 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] ~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).~ N/A
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/print/function-parameters.js [hunk 1]
```diff
@@ -223,8 +223,16 @@ function shouldGroupFunctionParameters(functionNode, returnTypeDoc) {
   );
 }
 
+function shouldBreakFunctionParameters(functionNode) {
+  const { params } = functionNode;
```
Comment (fisker) [inline src/language-js/print/function-parameters.js:227]: please use `getFunctionParameters`

Comment (seiyab) [reply src/language-js/print/function-parameters.js]: I changed it. Thank you for your review.

**Comment by fisker** [comment]
Changed base to `next` branch.

File: src/language-js/print/function-parameters.js [hunk 2]
```diff
@@ -223,8 +223,16 @@ function shouldGroupFunctionParameters(functionNode, returnTypeDoc) {
...
+  return (
+    params.length > 1 && params.some((p) => p.type === "TSParameterProperty")
```
[the remark this answers is not in the document]
Comment (fisker) [reply src/language-js/print/function-parameters.js]: I think it's fine to stay inline with one parameter.

**Review by fisker — APPROVED** [review]
LGTM, thank!

**Review by sosukesuzuki — APPROVED** [review]
I've updated snapshots file, LGTM

**Comment by danriedl** [comment]
Hi, thanks for that PR 🚀 
Is it already on a released version? 
I can see that it had been merged into `prettier:next` but can't find that branch.

**Comment by seiyab** [comment]
I think it is not released yet.
It will be released at V3.
It will take more time than usual since V3 has breaking changes.

**Comment by seiyab** [comment]
Or if you can't wait, alpha version (`3.0.0-alpha.11`) might work. Maybe.

**Comment by tutok** [comment]
is there any configuration flag which allows bringing back the previous formatting style?

**Pull request merged**